hedyorg/hedy
# 🪲 Show unsubmit button to teachers only for programs of their current students

This PR makes 3 changes:

1. Until now the checkbox for approving student adventures was only shown to the primary teacher. Now the checkbox is also shown to secondary teachers.
2. The unsubmit button is only displayed if the logged-in user is a current teacher (primary or secondary) of the student who submitted the program.
3. The statement that the program is submitted and cannot be altered together with the button to unsubmit have been moved, so that they do not take the upper central part of the screen.

Fixes #5990

**How to test**

Test user permissions

  - Do you agree that a secondary teacher of the class should be able to unsubmit and/or approve submitted programs?
  - A primary teacher can unsubmit the program of their student: log in as a student (student1) enrolled in a class and submit a program. Then log in as the primary teacher (teacher1) and check that you can unsubmit that program, check that you cannot unsubmit a program of the same student which has not been submitted.
  - A secondary teacher can unsubmit the program of their student: log in as a student (student1) enrolled in a class and submit a program. Then log in as the secondary teacher (e.g. teacher4) and check that you can unsubmit that program.
  - A teacher cannot unsubmit the program of a student who is not enrolled in their class: log in as a student (student1) enrolled in a class and submit a program. Then log in a teacher who is not a teacher of the student's class (teacher2) and check that you cannot unsubmit that program. Note that you should be able to see if a program is submitted and when.
  - A student cannot unsubmit the program of another student: log in as another student (student2) and check that you cannot unsubmit that program. Again, you should be able to see if a program is submitted and when

Check whether you agree with the layout. Please note that this page uses too many headings, different styles and overall does not look stylistically coherent. I intentionally made the minimal changes in the layout without changing styling.

This is how a submitted program looks if you are a teacher of the student:
<img width="1279" alt="Screenshot 2024-12-06 at 12 01 53" src="https://github.com/user-attachments/assets/5e66bfe8-4457-479c-81aa-f9a6efeece77">

And this is how if looks when you are another student or a teacher who is not in the student's class:
<img width="1270" alt="Screenshot 2024-12-06 at 12 02 20" src="https://github.com/user-attachments/assets/3b419210-dd29-4984-b43d-12bd33e402aa">
